library/asn1write.c

```diff
@@ -253,34 +253,37 @@ int mbedtls_asn1_write_int( unsigned char **p, unsigned char *start, int val )
     return( (int) len );
 }
 
-int mbedtls_asn1_write_printable_string( unsigned char **p, unsigned char *start,
-                                 const char *text, size_t text_len )
+int mbedtls_asn1_write_any_string( unsigned char **p, unsigned char *start, int tag,
```

Comment:
I don't like the name `mbedtls_asn1_write_any_string` - what about `mbedtls_asn1_write_tagged_string`, reflecting the flexibility in the tag?

Reply:
`mbedtls_asn1_write_tagged_string` sounds nice.